styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -387,4 +345,79 @@ static ResourceWithInstance create(String resource, InstanceState instanceState)
       return new AutoValue_Scheduler_ResourceWithInstance(resource, instanceState);
     }
   }
+
+  public static class SchedulerUtil {
```

Comment:
This refactoring feels weird. I think we can might just move it out completely from `Scheduler`. `Scheduler.SchedulerUtil` usage in `StyxScheduler` creates unnecessary dependency.